packages/path_provider/path_provider_linux/lib/src/path_provider_linux_stub.dart

```diff
@@ -0,0 +1,20 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:path_provider_platform_interface/path_provider_platform_interface.dart';
+
+/// A stub implementation to satisfy compilation of multi-platform packages that
+/// depend on path_provider_linux. This should never actually be created.
+///
+/// Notably, because path_provider needs to manually register
+/// path_provider_linux, anything with a transitive dependency on
+/// path_provider will also depend on path_provider_linux, not just at the
+/// pubspec level but the code level.
+class PathProviderLinux extends PathProviderPlatform {
+  /// Errors on attempted instantiation of the stub. It exists only to satisfy
+  /// compile-time dependencies, and should never actually be created.
+  PathProviderLinux() {
+    assert(false);
+  }
```

Comment:
The stub has to have everything the real version has, or you create analyzer errors, because the analyzer doesn't really understand conditional includes (and in fact it's not even clear how it *should* handle them)

I highly recommend that you make a utility file that has just the `getApplicationId` function, and only put that behind a conditional include so that there aren't two copies of things to maintain.